File: evaluating_platform/internal/model/asset.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type AssetType string

const (
	AssetTypeToolConfig AssetType = "tool_config"
	AssetTypeSuite      AssetType = "suite"
)

type AssetVisibility string

const (
	VisibilityPrivate AssetVisibility = "private"
	VisibilityOrg     AssetVisibility = "org"
	VisibilityPublic  AssetVisibility = "public"
)

type AssetStatus string

const (
	AssetStatusDraft      AssetStatus = "draft"
	AssetStatusTesting    AssetStatus = "testing"
	AssetStatusPublished  AssetStatus = "published"
	AssetStatusDeprecated AssetStatus = "deprecated"
)

type Asset struct {
	ID          uuid.UUID       `json:"id" db:"id"`
	ExpertID    uuid.UUID       `json:"expert_id" db:"expert_id"`
	Name        string          `json:"name" db:"name"`
	Description string          `json:"description" db:"description"`
	Type        AssetType       `json:"type" db:"type"`
	Visibility  AssetVisibility `json:"visibility" db:"visibility"`
	Status      AssetStatus     `json:"status" db:"status"`
	Version     string          `json:"version" db:"version"`
	Config      WorkflowConfig  `json:"config" db:"-"`
	PriceUnit   float64         `json:"price_unit" db:"price_unit"`
	CallCount   int64           `json:"call_count" db:"call_count"`
	CreatedAt   time.Time       `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at" db:"updated_at"`
}

// WorkflowConfig remains as a generic JSON config shape used by stored assets.
type WorkflowConfig struct {
	Nodes  []WorkflowNode         `json:"nodes"`
	Edges  []WorkflowEdge         `json:"edges"`
	Params map[string]interface{} `json:"params,omitempty"`
}

type WorkflowNode struct {
	ID       string                 `json:"id"`
	ToolName string                 `json:"tool_name"`
	Label    string                 `json:"label"`
	Params   map[string]interface{} `json:"params,omitempty"`
	Position NodePosition           `json:"position"`
}

type WorkflowEdge struct {
	ID     string `json:"id"`
	Source string `json:"source"`
	Target string `json:"target"`
}

type NodePosition struct {
	X float64 `json:"x"`
	Y float64 `json:"y"`
}

File: evaluating_platform/frontend/src/services/asset.ts
import api from './api'

export interface AssetNode {
  id: string
  tool_name: string
  label: string
  params?: Record<string, unknown>
  position: { x: number; y: number }
}

export interface AssetEdge {
  id: string
  source: string
  target: string
}

export interface AssetConfig {
  nodes: AssetNode[]
  edges: AssetEdge[]
  params?: Record<string, unknown>
}

export interface Asset {
  id: string
  expert_id: string
  name: string
  description: string
  type: 'tool_config' | 'suite'
  visibility: 'private' | 'org' | 'public'
  status: 'draft' | 'testing' | 'published' | 'deprecated'
  version: string
  config: AssetConfig
  price_unit: number
  call_count: number
  created_at: string
  updated_at: string
}

export interface CreateAssetData {
  name: string
  description?: string
  type: 'tool_config' | 'suite'
  visibility?: string
  version?: string
  config?: AssetConfig
  price_unit?: number
}

export const assetService = {
  async listPublic(type?: string, limit = 20, offset = 0): Promise<{ items: Asset[]; total: number }> {
    const res = await api.get('/tools', { params: { type, limit, offset } })
    return res.data
  },

  async listMine(limit = 50, offset = 0): Promise<{ items: Asset[]; total: number }> {
    const res = await api.get('/assets', { params: { mine: 1, limit, offset } })
    return res.data
  },

  async create(data: CreateAssetData): Promise<{ asset_id: string }> {
    const res = await api.post('/assets', data)
    return res.data
  },

  async deprecate(id: string): Promise<void> {
    await api.put(`/tools/${id}/deprecate`)
  },
}
